### PR TITLE
Merge 2.8.x into main

### PR DESCRIPTION
**Pull request by rasabot**
:auto_rickshaw: This PR should be merged automatically once it has been approved. If it doesn't happen:
- [ ] Handle merge conflicts
- [ ] Fix build errors

:bulb: It has been opened automatically after changes were merged in a feature branch.

### CODE REVIEW

**Review by alwx — APPROVED** [review]

**Comment by samsucik** [comment]
@dakshvar22 since I've made small changes on this branch (in order to resolve a merge conflict after merging `main` into this branch), could _you_ do the review here?

I had to re-generate the Poetry lock file and fix a tiny conflict in `memoization.py` (because of Johannes's work on speeding up AugMemo and Enable's work on turning components into graph ones). Everything else merged cleanly.

File: poetry.lock [hunk 1]
```diff
@@ -575,7 +568,7 @@ python-versions = "*"
 
 [[package]]
 name = "docspec"
-version = "1.0.1"
+version = "1.1.0"
```
Comment (wochinge) [inline poetry.lock:571]: 💡

File: rasa/core/brokers/sql.py [hunk 2]
```diff
@@ -5,6 +5,8 @@
 from typing import Any, Dict, Optional, Text, Generator
 
 from sqlalchemy.orm import Session
+from sqlalchemy.ext.declarative import declarative_base, DeclarativeMeta
+from sqlalchemy import Column, Integer, String, Text
```
Comment (samsucik) [inline rasa/core/brokers/sql.py]: @wochinge I've renamed this `Text` because it clashed with `typing.Text`. I hope it's okay this way.

Comment (wochinge) [reply rasa/core/brokers/sql.py]: Sure 👍🏻 I think using the full module is bit preferable to renaming (e.g. referring to it as `sqlachemy.Text`) but fine for now

**Pull request merged**